OpenRA.Mods.Common/Widgets/Logic/Ingame/GameInfoObjectivesLogic.cs

```diff
@@ -64,11 +64,15 @@ void PopulateObjectivesList(MissionObjectives mo, ScrollPanelWidget parent, Cont
 
 			foreach (var o in mo.Objectives.OrderBy(o => o.Type))
 			{
-				var objective = o; // Work around the loop closure issue in older versions of C#
-				var widget = template.Clone();
+				// Work around the loop closure issue in older versions of C#
+				var objective = o;
+
+				if (objective.Hidden && objective.State == ObjectiveState.Incomplete)
```

Comment:
I'm thinking of the GDI06 case now. Do we need the ability to hide an objective even after it is completed? That would also mean suppressing the notifications for it, for example.

In fact I would love if this PR would use GDI06 as a test-case instead of a throw-away, because that mission is the main motivation to have this in the first place.

Reply:
This does not make sense to me. If you want completed objectives hidden you might as well use boolean variables. What use case do you have in mind for objectives that never appear?

Reply:
Do we need hidden objectives at all? Why can't we add the objective and then immediately complete/fail it when the necessary condition (probably a trigger) is set?

Reply:
The case coming to my mind is having a completed primary objective, but the mission not ending yet. On the other hand, I think this can be worked around now that the type names are arbitrary. I'll remove the hidden type if nobody has a specific case for it.

Reply:
We probably need to clarify / argue about what the desired behaviour for GDI06 actually is.

IMO it's confusing and wrong to have an objective "Destroy the Nod ********." marked as failed if there is no way for the player to know what ******** actually is during the mission.

We could implement a new API to replace the objective (or maybe just its text) so that ******** becomes Airfield, but i'm not sure that this is actually something that we want?